worlds/generic/docs/setup_en.md

```diff
@@ -47,39 +50,142 @@ slot name and, if the relevant game requires it, files to associate it with that
+which you can create a room, and then [connect](#connecting).
+
+If you have downloaded the settings, or have created a settings file manually, this file can be uploaded on the
+[Generation Page](/generate) where you can also set any specific hosting settings.
+
+#### On your local installation
+
+To generate a game on your local machine, make sure to install the Archipelago software, and ensure to select the
+Generate component, as well as ROM setup, if necessary. Navigate to your Archipelago installation (usually 
+C:\ProgramData\Archipelago), and place the settings file you have either created or downloaded from the website in the
+`Players` folder.
+
+Run `ArchipelagoGenerate.exe`, and it will inform you whether the generation was successful or not. If successful, there
+will be an output zip in the `output` folder (usually named something like `AP_XXXXX.zip`). This will contain all
+relevant information to the session including the spoiler if one was generated.
+
+### Generating a multiplayer game
+
+Archipelago is a multi-game multiworld architecture, so any number of players and any number of games may be used to
+generate. Of note, the website currently limits to a maximum player count of 30. If you would like to generate a game
+larger than that, it must be done on the local installation. Generally, it is better to generate locally to free server
+resources, and host the resulting multiworld on the website.
+
+#### Gather All Player YAMLs
 
 All players that wish to play in the generated multiworld must have a YAML file which contains the settings that they
-wish to play with.
+wish to play with. One person should gather all files from all participants in the generated multiworld. It is possible
+for a single player to have multiple games, or even multiple slots of a single game, but each YAML must have a unique
+player name.
+
+#### On the website
+
+Gather all player YAML files into a single place, and compress them into a zip file. This can be done by pressing
+ctrl/cmd + clicking on each file until all are selected, right-clicking one of the files, and clicking
+`compress to ZIP file` or `send to > compressed folder`.
 
-Typically, a single participant of the multiworld will gather the YAML files from all other players. After getting the
-YAML files of each participant for your multiworld game they can be compressed into a ZIP folder to then be uploaded to
-the multiworld generator page. Multiworld generator
-page: [Archipelago Seed Generator Page](/generate)
+Navigate to the [Generate Page](/generate), select the host settings you would like, click on `Upload File`, and
+select the newly created zip from the opened window.
 
-#### Rolling a YAML Locally
+After some time, you will be redirected to a seed info page that will display the generated seed, the time it was
+created, the number of players, the spoiler (if one was created) and all rooms created from this seed.
+
+
+#### On your local installation
```

Comment:
It may be worthwhile to remove the word "roll" from this section and the next as people may not be familiar with it. They could be replaced with "generate" which is used elsewhere in the document.